src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -2204,11 +2207,15 @@ internal override void IntersectWithEnumerable(IEnumerable<T> other)
                     if (Contains(item))
                     {
                         toSave.Add(item);
-                        Remove(item);
                     }
                 }
-                Clear();
-                AddAllElements(toSave);
+
+                if (toSave.Count < Count)
+                {
+                    Clear();
+                    AddAllElements(toSave);
+                }
+
```

Comment:
Maybe it used to differ but the different parts were removed? Or someone really wanted that debug statement.

Using the parent implementation should be fine.